### PR TITLE
Add test native debug mode for Quarkus

### PR DESCRIPTION
**Pull request by kshpak**
**Pending state**. Awaiting the next version of Quarkus to be released.
TD for Quarkus native debug mode. The test is checked against  1.9.1.Final version of Quarkus

### CODE REVIEW

**Comment by rsvoboda** [comment]
run tests

File: testsuite/src/it/java/io/quarkus/ts/startstop/NativeDebugTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.concurrent.TimeUnit;
+
+import static io.quarkus.ts.startstop.utils.Commands.getQuarkusVersion;
+import static io.quarkus.ts.startstop.utils.Commands.getBuildCommand;
+import static io.quarkus.ts.startstop.utils.Commands.cleanTarget;
+import static io.quarkus.ts.startstop.utils.Commands.getBaseDir;
+import static io.quarkus.ts.startstop.utils.Logs.archiveLog;
+import static io.quarkus.ts.startstop.utils.Logs.checkLog;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
```
Comment (rsvoboda) [inline testsuite/src/it/java/io/quarkus/ts/startstop/NativeDebugTest.java:29]: No JavaDoc info about the test.

Also no info in README

File: testsuite/src/it/java/io/quarkus/ts/startstop/NativeDebugTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        } finally {
+            // Archive logs no matter what
+            archiveLog(cn, mn, buildLogA);
+            cleanTarget(app);
+        }
+    }
+
+    @Test
+    public void debugSymbolsCheck(TestInfo testInfo) throws IOException, InterruptedException {
+        testRuntime(testInfo, new File[]{
```
Comment (rsvoboda) [inline testsuite/src/it/java/io/quarkus/ts/startstop/NativeDebugTest.java:80]: Provide more context in the error message

```
[ERROR] debugSymbolsCheck{TestInfo}  Time elapsed: 155.958 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
	at io.quarkus.ts.startstop.NativeDebugTest.testRuntime(NativeDebugTest.java:67)
	at io.quarkus.ts.startstop.NativeDebugTest.debugSymbolsCheck(NativeDebugTest.java:80)
```
is not sufficient

**Review by rsvoboda — CHANGES_REQUESTED** [review]

**Comment by rsvoboda** [comment]
`mvn clean verify -Ptestsuite -Dtest=NativeDebugTest -Dquarkus.version=1.10.1.Final  -DincludeTags=native  -Dquarkus.native.container-build=true -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel:20.1-java11` is not working for me.

@kshpak what's your command?

**Comment by kshpak** [comment]
> `mvn clean verify -Ptestsuite -Dtest=NativeDebugTest -Dquarkus.version=1.10.1.Final -DincludeTags=native -Dquarkus.native.container-build=true -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel:20.1-java11` is not working for me.
> 
> @kshpak what's your command?

@rsvoboda I use ` mvn clean verify -Ptestsuite -DincludeTags=native -Dquarkus.version=1.9.1.Final  -Dquarkus.platform.version=1.9.1.Final -Dquarkus.native.container-build=true` gonna try your version

Working on changes

**Comment by rsvoboda** [comment]
@kshpak I don't see `quarkus-runner.debug` in target folder

**Comment by kshpak** [comment]
@rsvoboda that's interesting, for me it exists.
image

File: testsuite/src/it/java/io/quarkus/ts/startstop/NativeDebugTest.java (hunk 2 above)
Comment (kshpak) [reply testsuite/src/it/java/io/quarkus/ts/startstop/NativeDebugTest.java:86]: @rsvoboda PR updated

**Comment by rsvoboda** [comment]
I'm on macOS, tried the same on linux machine and .debug is available. Will file an issue for it.

File: testsuite/src/it/java/io/quarkus/ts/startstop/NativeDebugTest.java (hunk 2 above)
Comment (rsvoboda) [reply testsuite/src/it/java/io/quarkus/ts/startstop/NativeDebugTest.java:86]: better now

```
[ERROR] Failures:
[ERROR]   NativeDebugTest.debugSymbolsCheck:86->testRuntime:73 File or directory: /Users/rsvoboda/git/quarkus-startstop/app-jax-rs-minimal/target/quarkus-runner.debug is missing. ==> expected: <true> but was: <false>
[INFO]
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
```

**Review by rsvoboda — APPROVED** [review]

**Comment by rsvoboda** [comment]
https://github.com/quarkusio/quarkus/issues/13518 => docs related change, nothing related to the native debug behavior
https://github.com/quarkusio/quarkus/issues/13517 => maybe docs or info logging improvement

**Pull request merged**